game-core/src/integ_test/java/games/strategy/engine/lobby/server/db/EmailLimitIntegrationTest.java

```diff
@@ -45,10 +46,10 @@ private static String getStringWithLength(final int length) {
 
   private static void createAccountWithEmail(final String email) throws SQLException {
     try (PreparedStatement ps =
-        connection.prepareStatement("insert into ta_users (username, email, password) values (?, ?, ?)")) {
+        connection.prepareStatement("insert into ta_users (username, email, bcrypt_password) values (?, ?, ?)")) {
       ps.setString(1, Util.createUniqueTimeStamp());
       ps.setString(2, email);
-      ps.setString(3, games.strategy.util.Md5Crypt.crypt("password"));
+      ps.setString(3, BCrypt.hashpw("password", BCrypt.gensalt()));
```

Comment:
Does this change hashed pass from md5 to bcrypt? If so, how will migration go?

Reply:
This is just a test focused on the `email` column.  We are required to insert a value in one of the `password` or `bcrypt_password` columns due to the table constraints.  The type of password used is incidental to the test.

Reply:
Sorry not check, i wonder if it means if we can go simpler with a hardcoded string? Is the value from bcrypt special in any way?

Reply:
A bcrypt password hash is an encoded string with four fields (see [here](https://en.wikipedia.org/wiki/Bcrypt)).  So, yes, it is "special" in the sense that it can't just be an arbitrary string literal.

For this test to pass, it has to at least obey the following constraints based on the schema for this table (which are a subset of the full bcrypt syntax requirements):

* exactly 60 characters
* start with `$2a$`

Again, that knowledge seems incidental to the test.  We have test utilities scattered about that do things like generate usernames, hashed MACs, etc. for use in tests.  So, while I don't think a hard-coded bcrypt password hash would be appropriate here, we could extract a method to generate one (exactly the way it's being done above).  That way, at least the method name will make it clear to a reader what type of string is being used here.  (The `bcrypt_password` column name won't necessarily help a reader because it will probably be renamed to just `password` in the future when we remove support for MD5-crypt password hashes.)